File: optimal-partition-of-string/optimal-partition-of-string.js
//https://leetcode.com/problems/optimal-partition-of-string/description/


var partitionString = function(s) {
    let count = 0;
    const seen = new Set();
    seen.add(s[0])
    for (let i = 0; i < s.length; i++) {
        if (seen.has(s[i])) {
            seen.clear();
            count += 1
        }
        seen.add(s[i]);
    };
    return count
};

File: optimal-partition-of-string/optimal-partition-of-string.py
#https://leetcode.com/problems/optimal-partition-of-string/

class Solution:
    def partitionString(self, s: str) -> int:
        count = 0
        seen = set(s[0])
        for char in s:
            if char in seen:
                seen.clear()
                seen.add(char)
                count += 1
            seen.add(char)
        return count